code/game/objects/items/weapons/storage/belt.dm

```diff
@@ -12,9 +12,27 @@
 	equip_sound = 'sound/items/equip/toolbelt_equip.ogg'
 	/// Do we have overlays for items held inside the belt?
 	var/use_item_overlays = FALSE
+	/// Bypasses the "belt in bag" prevention if TRUE
+	var/storable = FALSE
+	/// Ignores update_weight() if TRUE
+	var/large = FALSE
+
+/obj/item/storage/belt/proc/update_weight()
+	if(large)
+		return
+	if(!length(contents) || storable)
+		w_class = WEIGHT_CLASS_NORMAL
+		return
+
+	w_class = WEIGHT_CLASS_BULKY
+
+/obj/item/storage/belt/Initialize(mapload)
+	. = ..()
+	update_weight()
 
 /obj/item/storage/belt/update_overlays()
 	. = ..()
+	update_weight()
```

Comment:
This feels a little out-of-place in `update_overlays()`, is there another proc called when things are added to/removed from storage that might be a bit better?

Reply:
This is already called in every other check for when new items are added to the belt. This makes it so that the change in look and the change in weight happen at the same time.